astropy/time/core.py

```diff
@@ -408,6 +409,40 @@ def now(cls):
 
     info = TimeInfo()
 
+    @classmethod
+    def strptime(cls, time_string, format_string):
```

Comment:
Make the call signature be:
```
def strptime(cls, format_spec, val, **kwargs):
```
Then pass `**kwargs` to the `cls` initializer later.  This allows specifying a different scale, precision, etc.

You will have to handle `format` specially if the user passes that in `kwargs`.  Something like:
```
format = kwargs.pop('format', None)
out = cls(*iterator.operands[1:], format='isot', **kwargs)
if format is not None:
    out.format = format

return out
```